src/pybind/cephfs/cephfs.pyx

```diff
@@ -2167,7 +2167,8 @@ cdef class LibCephFS(object):
                 ret = ceph_readlink(self.cluster, _path, buf, _size)
             if ret < 0:
                 raise make_ex(ret, "error in readlink")
-            return buf
+            py_bytes = buf[:ret]
+            return py_bytes
```

Comment:
why not just put
```py
return buf[:ret]
```
?

also, the title of the commit message reads like a bug report. ideally, the title of the commit message should use the imperative mood. see also https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes . please use `git commit --amend`  and `git push -f` for revising the commit message and its title.

Reply:
To my limited understanding (of Cython and the link Patrick provided), with the assignment of `py_bytes = ` , Cython converted `buf` to a python byte string rather than the original C string `buf`.

Reply:
i think both the explicit assignment of
```py
py_bytes = buf[:ret]
```
and the `return` statement
```
return buf[:ret]
```
copy the C string into a Python bytes string. FWIW, following is the cythonized code:
```c
    /* "cephfs.pyx":1986
 *             if ret < 0:
 *                 raise make_ex(ret, "error in readlink")
 *             return buf[:ret]             # <<<<<<<<<<<<<<
 *         finally:
 *             free(buf)
 */
    __Pyx_XDECREF(__pyx_r);
    __pyx_t_2 = __Pyx_PyBytes_FromStringAndSize(__pyx_v_buf + 0, __pyx_v_ret - 0); if (unlikely(!__pyx_t_2)) __PYX_ERR(1, 1986, __pyx_L4_error)
    __Pyx_GOTREF(__pyx_t_2);
    __pyx_r = __pyx_t_2;
    __pyx_t_2 = 0;
    goto __pyx_L3_return;
```

Reply:
Thank you for the verification.

Reply:
Rebuilt and tested fine.